crates/bpf-filtering/src/process_tree.rs

```diff
@@ -187,31 +190,36 @@ impl ProcessTree {
         namespaces: Namespaces,
         container_id: Option<ContainerId>,
     ) -> Result<&ProcessData, Error> {
-        let parent = self.processes.iter().find(|p| p.pid == ppid);
+        let parent = self.processes.get(&ppid);
         match parent {
             Some(parent) => {
                 let image = parent.image.to_string();
-
-                self.processes.push(ProcessData {
+                let process = ProcessData {
                     pid,
                     uid,
                     gid,
                     image,
                     parent: ppid,
                     namespaces,
                     container_id,
-                });
-                Ok(self.processes.last().unwrap())
+                };
+                self.processes.entry(pid).and_modify(|e| {
+                    warn!(
+                        "process tree key {pid} contained the process {e:?}, replacing it with a new process {process:?}"
+                    );
+                    *e = process.clone();
+                }).or_insert(process);
```

Comment:
maybe to avoid the `process.clone` you can probably match  on entry if occupied or vacant